test/remote/gateways/remote_securion_pay_test.rb

```diff
@@ -265,4 +339,30 @@ def test_card_declined
     assert_failure response
     assert_match Gateway::STANDARD_ERROR_CODE[:card_declined], response.error_code
   end
+
+  # remote test for shift4 V2 API
+
+  def test_successful_purchase_with_shift4_v2
```

Comment:
❓ Question:

Why `purchase` and `authorize` don't follow the `[@gateway, @gateway_shift4_v2]` pattern ?

Reply:
Good catch.